docs/book/src/component-config-tutorial/testdata/controller_manager_config.yaml

```diff
@@ -1,5 +1,5 @@
-apiVersion: controller-runtime.sigs.k8s.io/v1alpha1
-kind: ControllerManagerConfig
+apiVersion: config.tutorial.kubebuilder.io/v2
+kind: ProjectConfig
 metrics:
```

Comment:
We need to check this one.
It might not be the right way to move forward.

/hold

Reply:
Hi @tomasfabian,

Seems that confusion is because Kubebuilder is scaffolding the manifest by default but that will not work with the default scaffold and should only be done when the flag is used. Another part of the confusion might be because you are reading the end of the tutorial without following up it from the beginner, see: https://book.kubebuilder.io/component-config-tutorial/api-changes.html

In this way, I created an issue we dealing with: https://github.com/kubernetes-sigs/kubebuilder/issues/2782 and I hope that you do not mind if we close this PR. 

Thank you for bringing the topic and for your contribution.

Reply:
Hi @camilamacedo86,
I read the book from the beginning to the end. Isn't this a misunderstanding? I cloned the `kubebuilder` repo and tried to run  the provided ComponentConfig demo. 

I still think that the book is inconsistent, because it states that a ProjectConfig in GVK config.tutorial.kubebuilder.io/v2 is used:

> This type uses the new **ProjectConfig** kind under the GVK **config.tutorial.kubebuilder.io/v2**, with these custom configs we can add any yaml serializable fields that your controller needs and begin to reduce the reliance on flags to configure your project.

but instead of it uses:
```yaml
apiVersion: controller-runtime.sigs.k8s.io/v1alpha1
kind: ControllerManagerConfig
```

and therefore the provided demo throws the bellow exception during runtime execution (you can try it out and prove me wrong ;) ):
_could not decode file into runtime.Object_

I wanted to help you to improve the user experience, if you think that it is didactically and semantically correct then it is of course fine to close this PR.

Thanks, regards Tomas.